### ISSUE TITLE
New Webapp may be helpful to troubleshoot and get ip addresses

### ISSUE THREAD

**Issue by omriasta** [dlarrick/hass-kumo#55]
https://app.kumocloud.com/#/login

There is a new webapp. If you look in the dev tools of your browser, you can get a better idea of the communication with the backend and how it gets updates as well as the returned json. I wonder if those with missing ip addresses can see the ip using the webapp and maybe we can figure out how to get that to work for us. Maybe it would be easier to adapt to get the initial data to cache using the webapp...

**Comment by omriasta** [comment]
Not sure if this helps at all, taken from the javascript in the webapp:
```
        this.userWantsCelsius = !1,
        this.convert = function(e) {
            return null == e || "undefined" == typeof e ? null : this.userWantsCelsius ? Math.round(2 * e) / 2 : Math.round(1.8 * e + 32)
        }
        ,
        this.convertToBlewit = function(e) {
            return null == e || "undefined" == typeof e ? null : this.userWantsCelsius ? e : (e - 32) / 1.8
        }
        ,
        this.convertToCelsius = function(e) {
            return (e - 32) / 1.8
        }
        ,
        this.convertToFahrenheit = function(e) {
            return 1.8 * e + 32
        }
        ,
```

**Comment by dlarrick** [comment]
Yeah that's the normal way to convert F <-> C. I think the app is doing something different. Webapp might provide a bit more insight into some other things, though -- thanks, I'll take a look at some point.

**Comment by danielgoepp** [comment]
@dlarrick the web app does show this, it has a table:
https://drive.google.com/file/d/1CCXy7haUiOUvqXpl_AIOocPQ4OAvmyVr/view?usp=sharing
Starting at line 23914 (or just search for "-99")

**Comment by danielgoepp** [comment]
FYI, from Mitsubishi support:

"Also, if you are going to the app.kumocloud.com through a regular web browser, it’s a much older version and may not react like the version on a phone/tablet."

@omriasta is there still anything active here or should we close this out and continue our other threads on the IP address updates?

**Comment by omriasta** [comment]
I am closing though you may want to open a new issue with the C/F conversion table under pykumo

**Comment by brgaulin** [comment]
I confirmed that the web version is outdated or different compared to the phone app. I'm not seeing some settings on the web version that are available in the phone app such as LED settings on the v2 kumo units.

I extracted the android apk and it looks to be a cordova app, and I see the source js. It was mangled a bit, but it is similar. It is incredibly hard to read as many things were obfuscated or converted to hex. But if anyone needs it, note that apk files are actually jars, which are basically zips.
